### PR TITLE
Remove old code; Add Auto-generation of API docs

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [ ] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

**NOTE:** This PR is blocked and waiting for @binh-dam-ibigroup to resume API documentation work.

- Remove spring package
- Remove other dead code (e.g., websocket stuff)
- Move `middleware.spark.Main` to `middleware.OtpMiddlewareMain`
- Adds some swagger tests
- Introduces process for tracking/verifying API docs are up-to-date (maintained at `src/main/resources/doc.yaml` and auto-generating API Gateway enhanced documentation (`target/doc-api-gateway.yaml`) **Note: this has not been fully implemented! Needs some input from @binh-dam-ibigroup as to how to inject various info.**
- TODO: Modify controllers not configured with spark-swagger (perhaps we should handle in another PR if this gets approved).

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/OtpMiddlewareMain.java [hunk 1]
```diff
@@ -1,19 +1,17 @@
-package org.opentripplanner.middleware.spark;
+package org.opentripplanner.middleware;
 
 import com.beerboy.ss.SparkSwagger;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/OtpMiddlewareMain.java:6]: Remove unused imports.

File: src/main/java/org/opentripplanner/middleware/OtpMiddlewareMain.java [hunk 2]
```diff
@@ -76,18 +73,6 @@ private static void initializeHttpEndpoints() throws IOException {
...
-        spark.get("/sync", (req, res) -> BasicOtpDispatcher.executeRequestsInSequence());
-
-        // available at http://localhost:4567/async
-        spark.get("/async", (req, res) -> BasicOtpDispatcher.executeRequestsAsync());
-
-        // available at http://localhost:4567/plan
-        spark.get("/plan", OtpRequestProcessor::planning);
-
         // available at http://localhost:4567/api/secure/triprequests
         spark.get(API_PREFIX + "/secure/triprequests", TripHistoryController::getTripRequests);
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/OtpMiddlewareMain.java:77]: Not strictly part of this update, but stands out now that the other items have been removed. For consistency, this should be moved into a register method similar to `LogController.register`.

**Review by br648 — CHANGES_REQUESTED** [review]
Is the doc-api-gateway.yaml file generated manually? If so it's a bit onerous! Ignore this!

I think these files can be removed as well:

org.opentripplanner.middleware.BasicOtpDispatcher
org.opentripplanner.middleware.Const
org.opentripplanner.middleware.WebSocketClient

**Comment by binh-dam-ibigroup** [comment]
Closing this PR in favor of #47 and #50.

**Pull request closed without merging**